server/webapp/WEB-INF/rails/webpack/views/pages/elastic_profiles/elastic_profiles_widget.tsx

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2018 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Routes from "gen/ts-routes";
+import {MithrilComponent} from "jsx/mithril-component";
+import * as _ from "lodash";
+import * as m from "mithril";
+import {Stream} from "mithril/stream";
+import {ElasticProfile, ElasticProfiles} from "models/elastic_profiles/types";
+import {ExtensionType} from "models/shared/plugin_infos_new/extension_type";
+import {ElasticAgentSettings, Extension} from "models/shared/plugin_infos_new/extensions";
+import {PluginInfo} from "models/shared/plugin_infos_new/plugin_info";
+import * as Buttons from "views/components/buttons";
+import {ButtonIcon} from "views/components/buttons";
+import {CollapsiblePanel} from "views/components/collapsible_panel";
+import {FlashMessage, MessageType} from "views/components/flash_message";
+import {KeyValuePair} from "views/components/key_value_pair";
+import {ElasticProfileWidget} from "views/pages/elastic_profiles/elastic_profile_widget";
+import * as styles from "views/pages/elastic_profiles/index.scss";
+
+interface HasSuccessMessage {
+  successMessage: m.Children;
+  clearMessage: () => void;
+}
+
+type State = HasSuccessMessage;
+
+interface Attrs {
```

Comment:
See interfaces in `config_repos_widget.tsx`. Also — if you merge all these widget tsx files, you can share the same interface definitions instead of having to export them.

Reply:
Merged widget files.